File: src/database/DBCustomer.java
package database;

import model.Customer;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * DBDefaultDeliveryStop
 * Handles all database functionality for default delivery stops.
 *
 * @author Dennis Værum
 * @version 1.0
 * @since 21-05-15
 */

public class DBCustomer {
    private DBConnection dbConnection;
    private static DBCustomer instance;

    /**
     * Private constructor for singleton
     *
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    private DBCustomer() throws SQLException, ClassNotFoundException {
        dbConnection = DBConnection.getInstance();
    }

    /**
     * Singleton method for class
     *
     * @return the instance of DBCustomer
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    public static DBCustomer getInstance() throws SQLException, ClassNotFoundException {
        if (instance == null) {
            instance = new DBCustomer();
        }

        return instance;
    }

    /**
     * @param id from a DefaultDeliveryStop
     * @return list of all Customer for the given id
     */
    public ArrayList<Customer> getCustomers(long id) {
        ArrayList<Customer> list;
        String sql = String.format("select * from Customer where default_delivery_stop_id = '%s';", id);
        list = (ArrayList<Customer>) dbConnection.sendSQL(this, sql, "_formatCustomer");
        return list;
    }

    /**
     * @param rs takes the ResultSet from database
     * @return list of Customer
     */
    public ArrayList<Customer> _formatCustomer(ResultSet rs) {
        ArrayList<Customer> tableList = new ArrayList<>();
        try {
            while (rs.next()) {
                tableList.add(
                        new Customer(
                                rs.getLong("id"),
                                rs.getString("street_name"),
                                rs.getString("street_no"),
                                rs.getInt("zip_code"),
                                rs.getString("city"),
                                rs.getTime("time_of_delivery").toLocalTime()
                        ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }
}

File: Server/Database/DBCustomer.cs
using System;
using System.Collections.Generic;
using Model;
using System.Data;

namespace Server
{
    public class DBCustomer {
        private DBConnection dbConnection;
        private static DBCustomer instance;

        /// <summary>
        /// Gets the instance.
        /// </summary>
        /// <value>The instance.</value>
        public static DBCustomer Instance { 
            get { 
                if (instance == null)
                    instance = new DBCustomer();
                return instance;
            }
        }

        /// <summary>
        /// Initializes a new singleton of the <see cref="Server.DBCustomer"/> class.
        /// 
        private DBCustomer() {
            dbConnection = DBConnection.Instance;
        }

        /// <summary>
        /// Gets list of customers for database
        /// </summary>
        /// <returns>list of all Customer for the given id</returns>
        /// <param name="id">id from a DefaultDeliveryStop</param>
        public List<Customer> GetCustomers(long id) {
            List<Customer> list;
            String sql = String.Format("select * from Customer where default_delivery_stop_id = '{0}';", id);
            list = dbConnection.SendSQL<Customer>(sql, ConvertToCustomer);
            return list;
        }

        /// <summary>
        /// Formats the customer.
        /// </summary>
        /// <returns>The list of customer</returns>
        /// <param name="dataSet">The data set from database</param>
        private List<Customer> ConvertToCustomer(IDataReader dataSet) {
            List<Customer> tableList = new List<Customer>();
            while (dataSet.Read()) {
                tableList.Add(
                    new Customer(
                        dataSet.GetInt64(0),
                        dataSet.GetString(1),
                        dataSet.GetString(2),
                        dataSet.GetInt32(3),
                        dataSet.GetString(4),
                        dataSet.GetDateTime(5).ToLocalTime()
                    ));
            }
            return tableList;
        }
    }
}

